lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -672,9 +672,10 @@ static void buildModule(OpBuilder &builder, OperationState &result,
 }
 
 void FModuleOp::build(OpBuilder &builder, OperationState &result,
-                      StringAttr name, ArrayRef<PortInfo> ports,
-                      ArrayAttr annotations) {
+                      StringAttr name, ConventionAttr convention,
+                      ArrayRef<PortInfo> ports, ArrayAttr annotations) {
   buildModule(builder, result, name, ports, annotations);
+  result.addAttribute("convention", convention);
```

Comment:
Can we add "convention" attribute in `buildModule` function?

Reply:
This code is out here because mem modules don't have a settable convention attribute, but still build using the  buildModule helper. If mem modules grow a convention attr, we can share mode code, but I am punting on mem modules for this PR.

Reply:
There are equivalent concerns for mem modules, but the implementation is burried in annotations and multiple passes.  An issue, at least, should be filed for unifying memory module lowering with this.  (e.g. memories may flatten types, they may split up memories on field boundaries, etc).